packages/components/src/Checkbox/Checkbox.tsx

```diff
@@ -48,10 +44,26 @@ interface CheckboxProps {
    * Further description of the label
    */
   readonly description?: string;
+}
 
+interface CheckboxLabelProps extends BaseCheckboxProps {
+  /**
+   * Label that shows up beside the checkbox.
+   */
+  readonly label?: string;
   onChange?(newValue: boolean): void;
 }
 
+interface CheckboxChildrenProps extends BaseCheckboxProps {
+  /**
+   * Component children, which shows up as a label
+   */
+  readonly children?: JSX.Element;
+  onClick?(newValue?: boolean): void;
```

Comment:
What's the onclick for?

Reply:
It allows the link to be clicked without ticking the box. The link will have an onclick and will use `e.preventDefault()` to prevent the call from going up the chain

Reply:
Actually, I'm testing it again, it might not be necessary. I'll get back to you on that

Reply:
as @darryltec mentioned, It's a bit hard to tell the difference between `onChange` and `onClick` from the first glance. What do you think about renaming the `onClick` function to something like `onLinkClick`(something along that line) so that people can directly tell the difference between those two functions

Reply:
It ended up not needing an onClick, so I actually just ended up removing that!